cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -2344,22 +2343,10 @@ class SelectKUnstableMetaFunction : public MetaFunction {
     }
     switch (args[0].kind()) {
       case Datum::ARRAY: {
-        if (select_k_options.is_top_k()) {
-          return SelectKth<SortOrder::Descending>(*args[0].make_array(), select_k_options,
-                                                  ctx);
-        } else {
-          return SelectKth<SortOrder::Ascending>(*args[0].make_array(), select_k_options,
-                                                 ctx);
-        }
+        return SelectKth(*args[0].make_array(), select_k_options, ctx);
```

Comment:
Assumes that `args` is not empty? But can an `Array` be empty when it reaches this code? cc @lidavidm

Reply:
The arity is validated.